src/main/java/org/sagebionetworks/web/client/widget/table/explore/TableEntityWidgetV2.java

```diff
@@ -301,6 +302,20 @@ private void configureActions() {
           );
         }
       );
+    this.actionMenu.setActionListener(
+        Action.SHOW_GRID,
+        (action, e) -> {
+          boolean isVisible = !view.isSynapseGridVisible();
+          view.setSynapseGridVisible(isVisible);
+          // Toggle QueryWrapperPlotNav visibility - hide when grid is shown, show when grid is hidden
+          view.setQueryWrapperPlotNavVisible(!isVisible);
+          if (isVisible && currentQuery != null) {
+            view.configureSynapseGrid(currentQuery.getSql());
+          }
```

Comment:
Not sure how this will behave as you toggle between them but I suppose we can refine that behavior later.

I think maybe an imperative call to instantiate the grid could go here to avoid the `useEffect` to instantiate the session in the SynapseGrid component (via React's `useImperativeHandle`)